README.md

```diff
@@ -1,16 +1,45 @@
 [![Review Assignment Due Date](https://classroom.github.com/assets/deadline-readme-button-24ddc0f5d75046c5622901739e7c5dd533143b0c8e959d652212380cedb1ea36.svg)](https://classroom.github.com/a/FgMJElkj)
 # Theory vs. Practice
-
-- List 3 reasons why asymptotic analysis may be misleading with respect to
+## Task 1
+List 3 reasons why asymptotic analysis may be misleading with respect to
   actual performance in practice.
+### Sol
+  1) Constant Factors and Lower Order Terms. Really high constant factors or lower order terms that are ignored when simplifying for the sake of an asymptotic expression can cause very extreme time differences
+  2) Memory and cache. Inefficient memory utilization and cache misses can increase time significantly.
+  3) Distribution of Inputs and Special Cases. Already sorted, reverse sorted, or other special cases can drastically impact the efficiency and runtime.
+     
+## Task 2
+
+Suppose finding a particular element in a binary search tree with 1,000
+elements take 5 seconds. Given what you know about the asymptotic complexity
+of search in a binary search tree, how long would you guess finding the same
+element in a search tree with 10,000 elements takes? Explain your reasoning.
+
+### Sol
+The asymptotic complexity of search operations in a binary search tree is typically described as $O(log(n))$, where $n$ is the number of elements in the tree. This logarithmic complexity implies that the time required to search for any given element increases logarithmically with the number of elements in the tree. 
+
+To break this down further let's go step by step using the relation of 1000 elements in 5 seconds to solve a find operation, we can approximate the find operation for the same element in a tree with 10,000 elements. When the number of elements increases to 10,000 (an increase by a factor of 10), the logarithmic complexity means that the depth of the tree—and subsequently, the maximum number of comparisons needed—increases only slightly. If we calculate the depth of the tree using $n$ to represent the number of elements we get the following:
+- For $n = 1,000$, the depth is $log₂(1000) ≈ 9.966$.
+- For $n = 10,000$, the depth is $log₂(10000) = 13.288$.
+This is really cool because the depth increase from  roughly 10 to approximately 13.3 represents only a $33%$ increase in the depth of the tree, despite a $900%$ increase in the number of elements.
+Therefore, If 1,000 elements take 5 seconds, an increase to 10,000 elements would increase the time to $5 * (13.288/9.966) ≈ 6.33 seconds$, under the assumption that all other conditions remain constant and the tree remains balanced as an unbalanced tree would cause a much more dramatic search time.
```

Comment:
I'm not getting 6.33 seconds with the expression you're giving...

Reply:
Apologies. I added the terms instead of multiplying because I was in a rush. This is genuinely really embarrassing I will fix this and the other note you made